core/modules/czar/python/appInterface.py

```diff
@@ -101,14 +113,39 @@ def submitQuery(self, query, conditions):
         else:
             lock.unlock()
             return ("error","error",a.getError())
+
+        # Remember client context for kill-operations
+        proxyName = conditions["client_dst_name"]
+        proxyThread = conditions["server_thread_id"]
+        self._clientToServerId[(proxyThread, proxyName)] = a.sessionId
+
         return (resultName, lockName, "")
 
-    def killQuery(self, query, taskId):
+    def killQuery(self, sessionId):
         """Process a kill query command (experimental).  Params: query, taskId."""
```

Comment:
query, taskId should be replaced by sessionId. I'm of the opinion that repeating function arguments in comments without explanations is not useful, and then it's only useful if the meaning isn't clear from the name or there's other important information, like units.